CodePush.m

```diff
@@ -505,7 +506,12 @@ - (void)savePendingUpdate:(NSString *)packageHash
 RCT_EXPORT_METHOD(notifyApplicationReady:(RCTPromiseResolveBlock)resolve
                                 rejecter:(RCTPromiseRejectBlock)reject)
 {
-    [CodePush removePendingUpdate];
+    // We only mark a pending update as succeeded only if it has been booted up
+    // successfully, during which `_isFirstRunAfterUpdate` is set to true.
+    if (_isFirstRunAfterUpdate) {
```

Comment:
Are you sure that would create two instances? JS modules are supposed to be singletons already, so this pattern should work just fine. It would be good to verify that Babel / RN abide by this behavior, but I'm pretty sure they do.